tensorflow_addons/image/filters.py

```diff
@@ -112,6 +109,66 @@ def mean_filter2d(image, filter_shape=(3, 3), name=None):
         return y
 
 
+@tf.function
+def median_filter1d(image,filter_shape=3, name=None):
+    """This method performs 1D Median Filtering on images.Filter shape can be user given.
+       This method takes both kind of images where pixel values lie between 0 to 255 and where it lies between 0.0 and 1.0
+       Args:
+           input: A 3D `Tensor` of type `float32` or 'int32' or 'float64' or 'int64 and of shape`[rows, columns, channels]`
+
+           filter_shape: Optional Argument.Single Integer. Default value = 3
+
+        Returns:
+            A 1D median filtered image 3D tensor of shape [rows,columns,channels] and type 'int32'. Pixel value of returned 
+            tensor ranges between 0 to 255
+    """
+
+    with tf.name_scope(name or "median_filter1d"):
+        if not isinstance(filter_shape, int):
+            raise TypeError('Filter shape must be an Integer')
+        filter_shapex = filter_shape
+        (row, col, ch) = (image.shape[0], image.shape[1], image.shape[2])
+        if row != None and col != None and ch != None:
+            (row, col, ch) = (int(row), int(col), int(ch))
+        else:
+            raise TypeError('All the Dimensions of the input image '
+                            'tensor must be Integers.')
+        if row * col < filter_shapex:
+            raise ValueError(
+                'Number of Pixels in each dimension of the image should be \
+                more than the filter size. Got filter_shape %s' %
+                filter_shapex +
+                ' Image Shape (%s)' % image.shape)
+        if filter_shapex % 2 == 0 :
+            raise ValueError('Filter size should be odd. Got filter_shape '
+                             '(%s)' % filter_shapex)
+        image = tf.cast(image, tf.float32)
+        tf_i = tf.reshape(image, [row * col * ch])
+        ma = tf.math.reduce_max(tf_i)
+        image = _normalize(image, ma)
```

Comment:
Wonder why we have to normalize the image?

Reply:
In real life there are some images which are in the range of 0 to 255 and some images pixel values are in the floating range 0.0 to 1.0. Image viewers take care of both the images.  Hence,  my functanatilies also takes care of that.  And normalize is not a new addition in this pull request.  This is already present in the master file.

Reply:
So shouldn't it be:
- if input ranges from 0 to 1, then output ranges from 0 to 1.
- if input ranges 0~255, then output ranges 0 to 255.

:smile: ?

Reply:
I wanted uniformity in the output pattern, i.e, integer values.